### PR TITLE
Refactor how we get robot file

### PR DESCRIPTION
**Pull request by dionisislolas**
### What

Added functionality to read the robot file from an embedded file system.

### How to review

Changes ok. Tests pass. 

### Who can review

Anyone

### CODE REVIEW

File: robotseo/robot.go [hunk 1]
```diff
@@ -21,12 +22,27 @@ func Init(pathToRobotFile string) {
...
-		b, err = os.ReadFile(pathToRobotFile + fileName)
+
+		// if debug is true (the default) we get the robot file for the component tests
+		// otherwise we get the robot file from the local file store and the path specified by pathToRobotFile
+		if cfg.Debug {
+			b, err = features.GetRobotFile(fileName)
+		} else {
+			b, err = os.ReadFile(pathToRobotFile + fileName)
+		}
+
```
Comment (lindenmckenzie) [inline robotseo/robot.go]: why use debug and not the pathToRobotFile?

Comment (dionisislolas) [reply robotseo/robot.go]: Thank you Linden. This is now addressed.

File: features/robotassets.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package features
+
+import "embed"
+
+//go:embed steps/robot/robot_en.json
+//go:embed steps/robot/robot_cy.json
+
+var folder embed.FS
+
+func GetRobotFile(filename string) ([]byte, error) {
```
Comment (mr-nick17) [inline features/robotassets.go:11]: Missing go doc

File: features/robotassets.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	file, err := folder.ReadFile("steps/robot/" + filename)
+	if err != nil {
+		return nil, err
+	}
+	return file, nil
+}
```
Comment (mr-nick17) [inline features/robotassets.go:17]: Can this function be covered with a unit test?

File: robotseo/robot.go [hunk 4]
```diff
@@ -21,12 +22,21 @@ func Init(pathToRobotFile string) {
...
+	if !strings.HasSuffix(pathToRobotFile, "/") && pathToRobotFile != "" {
 		pathToRobotFile += "/"
 	}
 	for _, lang := range []config.Language{config.English, config.Welsh} {
 		fileName = "robot_" + lang.String() + ".json"
-		b, err = os.ReadFile(pathToRobotFile + fileName)
+
+		// if pathToRobotFile is empty (the default) we get the robot file for the component tests
+		// otherwise we get the robot file from the local file store and the path specified by pathToRobotFile
+		if pathToRobotFile == "" {
```
Comment (mr-nick17) [inline robotseo/robot.go:34]: Is this logic to fix the component test or something else? The comment suggests it's to fix tests but this package sits outside the tests?

**Review by mr-nick17 — COMMENTED** [review]
Some minor comments/questions. Suggest you rename the branch to follow git-flow as per our [standards](https://github.com/ONSdigital/dp/blob/main/guides/CONTRIBUTING.md#development-work)

File: features/robotassets.go (hunk 2 above)
Comment (dionisislolas) [reply features/robotassets.go:11]: Thank you Nick. This is now addressed.

File: features/robotassets.go (hunk 3 above)
Comment (dionisislolas) [reply features/robotassets.go:17]: Thank you Nick. This is now addressed.

File: robotseo/robot.go (hunk 4 above)
Comment (dionisislolas) [reply robotseo/robot.go:34]: Thank you Nick. This is now addressed.

**Review by mr-nick17 — APPROVED** [review]

**Pull request merged**